lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -414,23 +418,32 @@ def directives_include?(node, graphql_object, parent_type)
           true
         end
 
+        def set_all_interpreter_context(object, field, arguments, path)
+          if object
+            @context[:current_object] = @interpreter_context[:current_object] = object
+          end
+          if field
+            @context[:current_field] = @interpreter_context[:current_field] = field
+          end
+          if arguments
+            @context[:current_arguments] = @interpreter_context[:current_arguments] = arguments
+          end
+          if path
+            @context[:current_path] = @interpreter_context[:current_path] = path
+          end
+        end
```

Comment:
I found this work was a real hotspot, and this was the fastest I could come up with.